cmake/modules/RootMacros.cmake

```
@@ -2135,3 +2135,35 @@ function(generateManual name input output)
 
   install(FILES ${output} DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 endfunction()
+
+#----------------------------------------------------------------------------
+# If path is a system path, set the return variable is_system_path to true.
+# The 1st argument is the path that should be checked
+# The 2nd argument is the return value
+#----------------------------------------------------------------------------
+function (IS_SYSTEM_PATH path is_system_path)
+  foreach (dir ${CMAKE_SYSTEM_PATH})
+    if ("${path}" STREQUALS "$dir")
+      set(${is_system_path} true PARENT_SCOPE)
+    endif()
```

```suggestion
    return()
    endif()
```

Maybe my cmake is too rusty, but shouldn't we not always set is_system_path to false at the end of the function?

---

Indeed, I think the return is needed